nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-framework-cluster/src/main/java/org/apache/nifi/cluster/coordination/http/endpoints/ProcessorRunStatusDetailsEndpointMerger.java

```diff
@@ -0,0 +1,100 @@
+import java.util.ArrayList;
+import java.util.Map;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+public class ProcessorRunStatusDetailsEndpointMerger implements EndpointResponseMerger {
+    public static final String SCHEDULE_SUMMARY_URI = "/nifi-api/processors/run-status-details/queries";
+
+    @Override
+    public boolean canHandle(final URI uri, final String method) {
+        return "POST".equalsIgnoreCase(method) && SCHEDULE_SUMMARY_URI.equals(uri.getPath());
+    }
+
+    @Override
+    public NodeResponse merge(final URI uri, final String method, final Set<NodeResponse> successfulResponses, final Set<NodeResponse> problematicResponses, final NodeResponse clientResponse) {
+        if (!canHandle(uri, method)) {
+            throw new IllegalArgumentException("Cannot use Endpoint Mapper of type " + getClass().getSimpleName() + " to map responses for URI " + uri + ", HTTP Method " + method);
+        }
+
+        final ProcessorsRunStatusDetailsEntity responseEntity = clientResponse.getClientResponse().readEntity(ProcessorsRunStatusDetailsEntity.class);
+
+        // Create mapping of Processor ID to its schedule Summary.
+        final Map<String, ProcessorRunStatusDetailsEntity> scheduleSummaries = responseEntity.getRunStatusDetails().stream()
+            .collect(Collectors.toMap(entity -> entity.getRunStatusDetails().getId(), entity -> entity));
+
+        for (final NodeResponse nodeResponse : successfulResponses) {
+            final ProcessorsRunStatusDetailsEntity nodeResponseEntity = nodeResponse == clientResponse ? responseEntity :
+                nodeResponse.getClientResponse().readEntity(ProcessorsRunStatusDetailsEntity.class);
+
+            for (final ProcessorRunStatusDetailsEntity processorEntity : nodeResponseEntity.getRunStatusDetails()) {
+                final String processorId = processorEntity.getRunStatusDetails().getId();
+
+                final ProcessorRunStatusDetailsEntity mergedEntity = scheduleSummaries.computeIfAbsent(processorId, id -> new ProcessorRunStatusDetailsEntity());
+                merge(mergedEntity, processorEntity);
+            }
+        }
+
+        final ProcessorsRunStatusDetailsEntity mergedEntity = new ProcessorsRunStatusDetailsEntity();
+        mergedEntity.setRunStatusDetails(new ArrayList<>(scheduleSummaries.values()));
+        return new NodeResponse(clientResponse, mergedEntity);
+    }
+
+    private void merge(final ProcessorRunStatusDetailsEntity target, final ProcessorRunStatusDetailsEntity additional) {
+        PermissionsDtoMerger.mergePermissions(target.getPermissions(), additional.getPermissions());
+
+        final ProcessorRunStatusDetailsDTO targetSummaryDto = target.getRunStatusDetails();
+        final ProcessorRunStatusDetailsDTO additionalSummaryDto = additional.getRunStatusDetails();
+
+        // If name is null, it's because of permissions, so we want to nullify it in the target.
+        if (additionalSummaryDto.getName() == null) {
```

Comment:
Wouldn't `!additional.getPermissions().getCanRead()` be a bit more descriptive?

Reply:
It would be more descriptive in terms of the reasoning behind why it would be null. However, the convention is more to merge field-by-field. If for some reason the name were ever to become null for another reason, this would probably be safer.

Reply:
Yeah I think it's probably reasonable to use that as the conditional.